--- ../src-base/minecraft/org/bukkit/command/Command.java
+++ ../src-work/minecraft/org/bukkit/command/Command.java
@@ -31,7 +31,10 @@
     protected String usageMessage;
     private String permission;
     private String permissionMessage;
+    public co.aikar.timings.Timing timings; // Paper
+    public String getTimingName() {return getName();} // Paper
 
+
     protected Command(String name) {
         this(name, "", "/" + name, new ArrayList<String>());
     }
