--- ../src-base/minecraft/org/bukkit/event/player/PlayerFishEvent.java
+++ ../src-work/minecraft/org/bukkit/event/player/PlayerFishEvent.java
@@ -37,10 +37,13 @@
     }
 
     /**
-     * Gets the entity caught by the player
+     * Gets the entity caught by the player.
+     * <p>
+     * If player has fished successfully, the result may be cast to {@link
+     * Item}.
      *
-     * @return Entity caught by the player, null if fishing, bobber has gotten
-     *     stuck in the ground or nothing has been caught
+     * @return Entity caught by the player, Entity if fishing, and null if
+     *     bobber has gotten stuck in the ground or nothing has been caught
      */
     public Entity getCaught() {
         return entity;
