--- ../src-base/minecraft/org/bukkit/event/player/PlayerItemConsumeEvent.java
+++ ../src-work/minecraft/org/bukkit/event/player/PlayerItemConsumeEvent.java
@@ -20,6 +20,7 @@
     private static final HandlerList handlers = new HandlerList();
     private boolean isCancelled = false;
     private ItemStack item;
+    private ItemStack replacement; // Paper
 
     /**
      * @param player the player consuming
@@ -54,7 +55,29 @@
             this.item = item;
         }
     }
+    
+    // Paper start
+    /**
+     * Return the custom item stack that will replace the consumed item, or null if no
+     * custom replacement has been set (which means the default replacement will be used).
+     *
+     * @return The custom item stack that will replace the consumed item or null
+     */
+    public ItemStack getReplacement() {
+        return this.replacement;
+    }
 
+    /**
+     * Set a custom item stack to replace the consumed item. Pass null to clear any custom
+     * stack that has been set and use the default replacement.
+     *
+     * @param replacement Replacement item to set, null to clear any custom stack and use default
+     */
+    public void setReplacement(ItemStack replacement) {
+        this.replacement = replacement;
+    }
+    // Paper end
+
     public boolean isCancelled() {
         return this.isCancelled;
     }
