--- ../src-base/minecraft/net/minecraft/nbt/NBTTagString.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagString.java
@@ -72,4 +72,29 @@
     {
         return this.data;
     }
+
+    @Override
+    public String crucible_toString()
+    {
+        return quoteAndEscape(this.data);
+    }
+
+    public static String quoteAndEscape(String p_193588_0_)
+    {
+        StringBuilder stringbuilder = new StringBuilder("\"");
+
+        for (int i = 0; i < p_193588_0_.length(); ++i)
+        {
+            char c0 = p_193588_0_.charAt(i);
+
+            if (c0 == '\\' || c0 == '"')
+            {
+                stringbuilder.append('\\');
+            }
+
+            stringbuilder.append(c0);
+        }
+
+        return stringbuilder.append('"').toString();
+    }
 }
