--- ../src-base/minecraft/net/minecraft/nbt/NBTTagList.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagList.java
@@ -9,7 +9,7 @@
 
 public class NBTTagList extends NBTBase
 {
-    private List tagList = new ArrayList();
+    private List<NBTBase> tagList = new ArrayList<NBTBase>();
     private byte tagType = 0;
     private static final String __OBFID = "CL_00001224";
 
@@ -64,16 +64,19 @@
 
     public String toString()
     {
-        String s = "[";
-        int i = 0;
+        StringBuilder stringbuilder = new StringBuilder("[");
 
-        for (Iterator iterator = this.tagList.iterator(); iterator.hasNext(); ++i)
+        for (int i = 0; i < this.tagList.size(); ++i)
         {
-            NBTBase nbtbase = (NBTBase)iterator.next();
-            s = s + "" + i + ':' + nbtbase + ',';
+            if (i != 0)
+            {
+                stringbuilder.append(',');
+            }
+
+            stringbuilder.append(this.tagList.get(i));
         }
 
-        return s + "]";
+        return stringbuilder.append(']').toString();
     }
 
     public void appendTag(NBTBase p_74742_1_)
@@ -192,11 +195,11 @@
     {
         NBTTagList nbttaglist = new NBTTagList();
         nbttaglist.tagType = this.tagType;
-        Iterator iterator = this.tagList.iterator();
-
-        while (iterator.hasNext())
+        if ( nbttaglist.tagList instanceof ArrayList ) // Thermos, ensure we dont create arrays to then delete them
+        	((ArrayList<NBTBase>)nbttaglist.tagList).ensureCapacity(this.tagList.size());
+        
+        for(NBTBase nbtbase : this.tagList)
         {
-            NBTBase nbtbase = (NBTBase)iterator.next();
             NBTBase nbtbase1 = nbtbase.copy();
             nbttaglist.tagList.add(nbtbase1);
         }
