--- ../src-base/minecraft/net/minecraft/nbt/NBTTagEnd.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagEnd.java
@@ -1,10 +1,12 @@
 package net.minecraft.nbt;
 
+import io.github.crucible.nbt.ICrucibleString;
+
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class NBTTagEnd extends NBTBase
+public class NBTTagEnd extends NBTBase implements ICrucibleString
 {
     private static final String __OBFID = "CL_00001219";
 
@@ -26,4 +28,9 @@
     {
         return new NBTTagEnd();
     }
+
+    @Override
+    public String crucible_toString(){
+        return this.toString();
+    }
 }
