--- ../src-base/minecraft/net/minecraft/nbt/NBTTagString.java
+++ ../src-work/minecraft/net/minecraft/nbt/NBTTagString.java
@@ -1,10 +1,12 @@
 package net.minecraft.nbt;
 
+import io.github.crucible.nbt.ICrucibleString;
+
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
 
-public class NBTTagString extends NBTBase
+public class NBTTagString extends NBTBase implements ICrucibleString
 {
     private String data;
     private static final String __OBFID = "CL_00001228";
@@ -72,4 +74,29 @@
     {
         return this.data;
     }
+
+    @Override
+    public String crucible_toString()
+    {
+        return quoteAndEscape(this.data);
+    }
+
+    public static String quoteAndEscape(String p_193588_0_)
+    {
+        StringBuilder stringbuilder = new StringBuilder("\"");
+
+        for (int i = 0; i < p_193588_0_.length(); ++i)
+        {
+            char c0 = p_193588_0_.charAt(i);
+
+            if (c0 == '\\' || c0 == '"')
+            {
+                stringbuilder.append('\\');
+            }
+
+            stringbuilder.append(c0);
+        }
+
+        return stringbuilder.append('"').toString();
+    }
 }
