--- ../src-base/minecraft/cpw/mods/fml/common/launcher/FMLServerTweaker.java
+++ ../src-work/minecraft/cpw/mods/fml/common/launcher/FMLServerTweaker.java
@@ -19,6 +19,7 @@
         classLoader.addTransformerExclusion("cpw.mods.fml.repackage.");
         classLoader.addTransformerExclusion("cpw.mods.fml.relauncher.");
         classLoader.addTransformerExclusion("cpw.mods.fml.common.asm.transformers.");
+        classLoader.addTransformerExclusion("io.github.crucible.bootstrap."); // Crucible - make our bootstrap stuff visible for both obfuscated and not obfuscated sides
         classLoader.addClassLoaderExclusion("LZMA.");
         FMLLaunchHandler.configureForServerLaunch(classLoader, this);
         FMLLaunchHandler.appendCoreMods();
