--- ../src-base/minecraft/org/bukkit/inventory/meta/ItemMeta.java
+++ ../src-work/minecraft/org/bukkit/inventory/meta/ItemMeta.java
@@ -124,6 +124,34 @@
     */
     boolean hasConflictingEnchant(Enchantment ench);
 
+    // Spigot start
+    public class Spigot
+    {
+
+        /**
+         * Sets the unbreakable tag
+         *
+         * @param unbreakable true if set unbreakable
+         */
+        public void setUnbreakable(boolean unbreakable)
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
+
+        /**
+         * Return if the unbreakable tag is true
+         *
+         * @return true if the unbreakable tag is true
+         */
+        public boolean isUnbreakable()
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
+    }
+
+    Spigot spigot();
+    // Spigot end
+
     @SuppressWarnings("javadoc")
     ItemMeta clone();
 }
