--- ../src-base/minecraft/org/bukkit/configuration/serialization/ConfigurationSerializable.java
+++ ../src-work/minecraft/org/bukkit/configuration/serialization/ConfigurationSerializable.java
@@ -9,11 +9,11 @@
  * methods as defined by this interface:
  * <ul>
  * <li>A static method "deserialize" that accepts a single {@link Map}&lt;
- * {@link String}, {@link Object}> and returns the class.</li>
+ * {@link String}, {@link Object}&gt; and returns the class.</li>
  * <li>A static method "valueOf" that accepts a single {@link Map}&lt;{@link
- * String}, {@link Object}> and returns the class.</li>
+ * String}, {@link Object}&gt; and returns the class.</li>
  * <li>A constructor that accepts a single {@link Map}&lt;{@link String},
- * {@link Object}>.</li>
+ * {@link Object}&gt;.</li>
  * </ul>
  * In addition to implementing this interface, you must register the class
  * with {@link ConfigurationSerialization#registerClass(Class)}.
