--- ../src-base/minecraft/org/bukkit/scoreboard/Score.java
+++ ../src-work/minecraft/org/bukkit/scoreboard/Score.java
@@ -51,7 +51,18 @@
      */
     void setScore(int score) throws IllegalStateException;
 
+    // Spigot start
     /**
+     * Shows if this score has been set at any point in time.
+     * 
+     * @return if this score has been set before
+     * @throws IllegalStateException if the associated objective has been
+     *     unregistered
+     */
+    boolean isScoreSet() throws IllegalStateException;
+    // Spigot end
+
+    /**
      * Gets the scoreboard for the associated objective.
      *
      * @return the owning objective's scoreboard, or null if it has been
