--- ../src-base/minecraft/cpw/mods/fml/common/registry/ItemStackHolderRef.java
+++ ../src-work/minecraft/cpw/mods/fml/common/registry/ItemStackHolderRef.java
@@ -1,5 +1,7 @@
 package cpw.mods.fml.common.registry;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
@@ -40,20 +42,26 @@
     private static Object reflectionFactory;
     private static Method newFieldAccessor;
     private static Method fieldAccessorSet;
+    private static MethodHandle fieldSetter; // Crucible - lwjgl3ify field
     private static void makeWritable(Field f)
     {
         try
         {
-            if (modifiersField == null)
-            {
-                Method getReflectionFactory = Class.forName("sun.reflect.ReflectionFactory").getDeclaredMethod("getReflectionFactory");
-                reflectionFactory = getReflectionFactory.invoke(null);
-                newFieldAccessor = Class.forName("sun.reflect.ReflectionFactory").getDeclaredMethod("newFieldAccessor", Field.class, boolean.class);
-                fieldAccessorSet = Class.forName("sun.reflect.FieldAccessor").getDeclaredMethod("set", Object.class, Object.class);
-                modifiersField = Field.class.getDeclaredField("modifiers");
-                modifiersField.setAccessible(true);
-            }
-            modifiersField.setInt(f, f.getModifiers() & ~Modifier.FINAL);
+            // Crucible start - implement lwjgl3ify patch directly
+//            if (modifiersField == null)
+//            {
+//                Method getReflectionFactory = Class.forName("sun.reflect.ReflectionFactory").getDeclaredMethod("getReflectionFactory");
+//                reflectionFactory = getReflectionFactory.invoke(null);
+//                newFieldAccessor = Class.forName("sun.reflect.ReflectionFactory").getDeclaredMethod("newFieldAccessor", Field.class, boolean.class);
+//                fieldAccessorSet = Class.forName("sun.reflect.FieldAccessor").getDeclaredMethod("set", Object.class, Object.class);
+//                modifiersField = Field.class.getDeclaredField("modifiers");
+//                modifiersField.setAccessible(true);
+//            }
+//            modifiersField.setInt(f, f.getModifiers() & ~Modifier.FINAL);
+            f.setAccessible(true);
+            fieldSetter = MethodHandles.lookup()
+              .unreflectSetter(f);
+            // Crucible end
         } catch (Exception e)
         {
             throw Throwables.propagate(e);
@@ -62,23 +70,53 @@
 
     public void apply()
     {
+        // Crucible start - implement lwjgl3ify patch directly
+//        ItemStack is;
+//        try
+//        {
+//            is = GameRegistry.makeItemStack(itemName, meta, 1, serializednbt);
+//        } catch (RuntimeException e)
+//        {
+//            FMLLog.getLogger().log(Level.ERROR, "Caught exception processing itemstack {},{},{} in annotation at {}.{}", itemName, meta, serializednbt,field.getClass().getName(),field.getName());
+//            throw e;
+//        }
+//        try
+//        {
+//            Object fieldAccessor = newFieldAccessor.invoke(reflectionFactory, field, false);
+//            fieldAccessorSet.invoke(fieldAccessor, null, is);
+//        }
+//        catch (Exception e)
+//        {
+//            FMLLog.getLogger().log(Level.WARN, "Unable to set {} with value {},{},{}", this.field, this.itemName, this.meta, this.serializednbt);
+//        }
         ItemStack is;
-        try
-        {
+        try {
             is = GameRegistry.makeItemStack(itemName, meta, 1, serializednbt);
-        } catch (RuntimeException e)
-        {
-            FMLLog.getLogger().log(Level.ERROR, "Caught exception processing itemstack {},{},{} in annotation at {}.{}", itemName, meta, serializednbt,field.getClass().getName(),field.getName());
+        } catch (RuntimeException e) {
+            FMLLog.getLogger()
+              .log(
+                Level.ERROR,
+                "Caught exception processing itemstack {},{},{} in annotation at {}.{}",
+                itemName,
+                meta,
+                serializednbt,
+                field.getClass()
+                  .getName(),
+                field.getName());
             throw e;
         }
-        try
-        {
-            Object fieldAccessor = newFieldAccessor.invoke(reflectionFactory, field, false);
-            fieldAccessorSet.invoke(fieldAccessor, null, is);
+        try {
+            fieldSetter.invoke(is);
+        } catch (Throwable e) {
+            FMLLog.getLogger()
+              .log(
+                Level.WARN,
+                "Unable to set {} with value {},{},{}",
+                this.field,
+                this.itemName,
+                this.meta,
+                this.serializednbt);
         }
-        catch (Exception e)
-        {
-            FMLLog.getLogger().log(Level.WARN, "Unable to set {} with value {},{},{}", this.field, this.itemName, this.meta, this.serializednbt);
-        }
+        // Crucible end
     }
 }
