--- ../src-base/minecraft/net/minecraft/enchantment/Enchantment.java
+++ ../src-work/minecraft/net/minecraft/enchantment/Enchantment.java
@@ -7,6 +7,7 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.StatCollector;
+import org.bukkit.craftbukkit.v1_7_R4.enchantments.CraftEnchantment;
 
 public abstract class Enchantment
 {
@@ -56,6 +57,8 @@
         {
             enchantmentsList[p_i1926_1_] = this;
         }
+
+        org.bukkit.enchantments.Enchantment.registerEnchantment(new CraftEnchantment(this)); // CraftBukkit
     }
 
     public int getWeight()
