--- ../src-base/minecraft/net/minecraft/command/server/CommandSummon.java
+++ ../src-work/minecraft/net/minecraft/command/server/CommandSummon.java
@@ -1,6 +1,8 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
+import io.github.crucible.nbt.Crucible_JsonToNBT;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
@@ -8,7 +10,6 @@
 import net.minecraft.entity.EntityList;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.IEntityLivingData;
-import net.minecraft.nbt.JsonToNBT;
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.nbt.NBTException;
 import net.minecraft.nbt.NBTTagCompound;
@@ -71,7 +72,7 @@
 
                     try
                     {
-                        NBTBase nbtbase = JsonToNBT.func_150315_a(ichatcomponent.getUnformattedText());
+                        NBTBase nbtbase = Crucible_JsonToNBT.getTagFromJson(ichatcomponent.getUnformattedText());
 
                         if (!(nbtbase instanceof NBTTagCompound))
                         {
