--- ../src-base/minecraft/cpw/mods/fml/relauncher/ServerLaunchWrapper.java
+++ ../src-work/minecraft/cpw/mods/fml/relauncher/ServerLaunchWrapper.java
@@ -1,5 +1,7 @@
 package cpw.mods.fml.relauncher;
 
+import io.github.crucible.CrucibleServerMainHook;
+
 import java.lang.reflect.Method;
 
 public class ServerLaunchWrapper {
@@ -9,6 +11,7 @@
      */
     public static void main(String[] args)
     {
+        CrucibleServerMainHook.relaunchMain(args);
         new ServerLaunchWrapper().run(args);
     }
 
